### PR TITLE
feat: (IAC-750) Add type and description for all the variables in the variables.tf files

### PR DESCRIPTION
**Pull request by riragh**
# Changes:
Updated all the `variables.tf` files in this repo to have description and type.

# Tests:
Cluster creation successful, see internal ticket for details

### CODE REVIEW

File: modules/azure_aks/variables.tf [hunk 1]
```diff
@@ -1,81 +1,123 @@
...
   validation {
-    condition     = contains(["Free", "Paid"],  var.aks_cluster_sku_tier)
+    condition     = contains(["Free", "Paid"], var.aks_cluster_sku_tier)
     error_message = "ERROR: Valid types are \"Free\" and \"Paid\"!"
   }
 }
 
 variable "aks_private_cluster" {
-  default = false
+  description = "The cluster API endpoint uses Private IP address?"
```
Comment (dhoucgitter) [inline modules/azure_aks/variables.tf:39]: description seems unclear and ends with a question mark

File: modules/aks_node_pool/variables.tf [hunk 2]
```diff
@@ -1,28 +1,36 @@
...
 variable "vnet_subnet_id" {
-  default = null
+  description = " (Optional) The ID of the Subnet where this Node Pool should exist. Changing this forces a new resource to be created."
+  type        = string
+  default     = null
 }
 
 variable "machine_type" {
-  type = string
+  description = " (Required) The SKU which should be used for the Virtual Machines used in this Node Pool. Changing this forces a new resource to be created."
```
Comment (dhoucgitter) [inline modules/aks_node_pool/variables.tf:26]: just a formatting question, noticed a leading space for these 3 descriptions

File: modules/azure_aks/variables.tf [hunk 3]
```diff
@@ -1,81 +1,123 @@
...
 variable "aks_cluster_node_vm_size" {
-  default = "Standard_D4_v2"
+  description = "The default virtual machine size for the Kubernetes agents"
+  type        = string
+  default     = "Standard_D4_v2"
 }
 
 variable "aks_cluster_node_admin" {
-  default = "ubuntu"
+  description = "The operating system Admin User for VMs of AKS cluster nodes"
```
Comment (dhoucgitter) [inline modules/azure_aks/variables.tf:64]: suggestion: use AKS cluster nodes in both descriptions, Kubernetes agents seems like a different object but looks they both refer to AKS cluster nodes.

File: modules/azurerm_vm/variables.tf [hunk 4]
```diff
@@ -1,115 +1,149 @@
...
+variable "data_disk_count" {
+  description = "Number of disks to be created"
+  type        = number
+  default     = 0
 }
 
-variable data_disk_size {
-  default = 128
+variable "data_disk_size" {
+  description = "Specifies the size of the managed disk to create in gigabytes."
```
Comment (dhoucgitter) [inline modules/azurerm_vm/variables.tf:86]: noticed that GB was used in other locations but spelled out here, if it makes sense could use the same for all.

File: modules/azurerm_vm/variables.tf [hunk 5]
```diff
@@ -1,115 +1,149 @@
...
+variable "cloud_init" {
+  description = "cloud_init config"
+  type        = string
+  default     = ""
 }
 
-variable create_public_ip {
-  default = false
+variable "create_public_ip" {
+  description = "Reference to a Public IP Address to associate with this NIC"
```
Comment (dhoucgitter) [inline modules/azurerm_vm/variables.tf:140]: does this boolean also create a public IP address or just reference one that already exists, not sure given the text here?

File: variables.tf [hunk 6]
```diff
@@ -88,87 +103,133 @@ variable "postgres_public_access_cidrs" {
...
+
+  validation {
+    condition     = contains(["kubenet", "azure"], var.aks_network_plugin)
+    error_message = "Error: Currently the supported values are `kubenet` and `azure`"
+  }
 }
 
 variable "aks_network_policy" {
-  description = "Sets up network policy to be used with Azure CNI. Network policy allows us to control the traffic flow between pods. Currently supported values are calico and azure. Changing this forces a new resource to be created."
+  description = "Sets up network policy to be used with Azure CNI. Network policy allows to control the traffic flow between pods. Currently supported values are calico and azure. Changing this forces a new resource to be created."
```
Comment (dhoucgitter) [inline variables.tf:160]:
```suggestion
  description = "Sets up network policy to be used with Azure CNI. Network policy allows control of the traffic flow between pods. Currently supported values are calico and azure. Changing this forces a new resource to be created."
```
Consider asking Susan Pearsall to review all these new descriptions.

File: modules/azure_aks/variables.tf (hunk 1 above)
Comment (riragh) [reply modules/azure_aks/variables.tf]: Updated the description in the latest commit.

File: modules/aks_node_pool/variables.tf (hunk 2 above)
Comment (riragh) [reply modules/aks_node_pool/variables.tf]: Updated the descriptions in the latest commit.

File: modules/azure_aks/variables.tf (hunk 3 above)
Comment (riragh) [reply modules/azure_aks/variables.tf]: Updated the description in the latest commit.

File: modules/azurerm_vm/variables.tf (hunk 4 above)
Comment (riragh) [reply modules/azurerm_vm/variables.tf]: Updated the description in the latest commit.

File: modules/azurerm_vm/variables.tf (hunk 5 above)
Comment (riragh) [reply modules/azurerm_vm/variables.tf]: Updated the description in the latest commit.

File: modules/aks_node_pool/variables.tf [hunk 7]
```diff
@@ -97,10 +95,31 @@ variable "orchestrator_version" {
...
+#   description = "The Eviction Policy which should be used for Virtual Machines within the Virtual Machine Scale Set powering this Node Pool. Possible values are Deallocate and Delete. Changing this forces a new resource to be created. An Eviction Policy can only be configured when priority is set to Spot"
+#   type        = string
+#   default     = null
+# }
+
+# variable "spot_max_price" {
+#   description = "The maximum price you're willing to pay in USD per Virtual Machine. Valid values are -1 (the current on-demand price for a Virtual Machine) or a positive value with up to five decimal places. Changing this forces a new resource to be created."
+#   type        = number
+#   default     = -1
+# }
```
[the remark this answers is not in the document]
Comment (thpang) [reply modules/aks_node_pool/variables.tf:125]: Any info on this question?

**Review by thpang — APPROVED** [review]

**Review by dhoucgitter — APPROVED** [review]

**Pull request merged**